### PR TITLE
ES6 createScript 

### PR DESCRIPTION
**Pull request by kpal81xd**
- Modifies `scriptType` extension of `ScriptType` to use ES6 syntax

I confirm I have read the [contributing guidelines](https://github.com/playcanvas/engine/blob/master/.github/CONTRIBUTING.md) and signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

File: src/framework/script/script.js [hunk 1]
```diff
@@ -65,16 +64,13 @@ function createScript(name, app) {
     if (reservedScriptNames.has(name))
         throw new Error(`Script name '${name}' is reserved, please rename the script`);
 
-    const scriptType = function (args) {
-        EventHandler.prototype.initEventHandler.call(this);
-        ScriptType.prototype.initScriptType.call(this, args);
-    };
+    class scriptType extends ScriptType {}
```
Comment (willeastcott) [inline src/framework/script/script.js:67]: What about something like:

```
    class ScriptTypeWithAttributes extends ScriptType {
        attributes = new ScriptAttributes(this);
    }

    registerScript(ScriptTypeWithAttributes, name, app);
    return ScriptTypeWithAttributes;
    ```

Comment (kpal81xd) [reply src/framework/script/script.js:67]: I did try that but I got a linting error for using `this` as a parameter. Seems to work fine tho if I just ignore the linting error

Comment (kpal81xd) [reply src/framework/script/script.js]: Actually so does the existing method - The linting just didnt show before 👍

**Review by marklundin — APPROVED** [review]

File: src/framework/script/script.js [hunk 2]
```diff
@@ -65,19 +64,13 @@ function createScript(name, app) {
...
-        ScriptType.prototype.initScriptType.call(this, args);
-    };
-
-    scriptType.prototype = Object.create(ScriptType.prototype);
-    scriptType.prototype.constructor = scriptType;
-
-    scriptType.extend = ScriptType.extend;
-    scriptType.attributes = new ScriptAttributes(scriptType);
+    class ScriptTypeWithAttributes extends ScriptType {
+        // @ts-ignore
```
Comment (willeastcott) [inline src/framework/script/script.js:68]: Why is the `@ts-ignore` needed?

**Review by willeastcott — APPROVED** [review]

**Pull request merged**